crates/bifrost/src/providers/replicated_loglet/sequencer/appender.rs

```diff
@@ -149,7 +147,14 @@ impl<T: TransportConnect> SequencerAppender<T> {
                 State::Done | State::Cancelled | State::Sealed => break state,
                 State::Wave { graylist } => {
                     self.current_wave += 1;
-                    let Some(next_state) = cancellation
+                    // # Why is this cancellation safe?
+                    // Because we don't await any futures inside the join_next() loop, so we are
+                    // confident that we either have cancelled before resolving the commit token
+                    // or before. Even if a store was fully replicated and we cancelled before
```

Comment:
does the or case should refer to something else than "before"?

Reply:
Will fix the commend